hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerTask.java

```diff
@@ -341,6 +341,16 @@ public List<ContainerBalancerTaskIterationStatusInfo> getCurrentIterationsStatis
         .max()
         .orElse(0);
 
+    // the balancing thread could go to sleep without having initialized findTargetStrategy, so we check for null here
+    Map<DatanodeDetails, Long> sizeEnteringNodes = Collections.emptyMap();
+    if (findTargetStrategy != null) {
```

Comment:
There is non-obvious, that findTarget(Source)Strategy can be null. Maybe it will be better to use some default realisation with empty map, if org.apache.hadoop.hdds.scm.container.balancer.ContainerBalancerTask#initializeIteration was not executed?

Reply:
Not sure what you mean here. Are you suggesting initialising `findTargetStrategy` in the constructor of `ContainerBalancerTask` (like `findSourceStrategy`)? That should work as well.

Reply:
Cool! Yes, you are right. That's what I mean)